src/caffe/util/io.cpp

```diff
@@ -45,6 +47,13 @@ bool ReadProtoFromTextFile(const char* filename, Message* proto) {
   return success;
 }
 
+bool ReadProtoFromTextStream(std::istream& stream, Message* proto) {
+  IstreamInputStream* input = new IstreamInputStream(&stream);
```

Comment:
Why "new"? Not exception safe. Unnecessary heap allocation.

Reply:
fixed